src/main/java/org/broadinstitute/hellbender/utils/io/BlockCompressedIntervalStream.java

```diff
@@ -0,0 +1,612 @@
+package org.broadinstitute.hellbender.utils.io;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.seekablestream.SeekableBufferedStream;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.seekablestream.SeekableStreamFactory;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.tribble.*;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SVInterval;
+import org.broadinstitute.hellbender.utils.SVIntervalTree;
+import org.broadinstitute.hellbender.utils.codecs.FeatureSink;
+import org.broadinstitute.hellbender.utils.codecs.FeaturesHeader;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.*;
+
+public class BlockCompressedIntervalStream {
```

Comment:
Shouldn't `BlockCompressedIntervalStream` be covered by direct unit tests?

Reply:
Yes.  I thought the indirect tests covered everything, but they didn't.  Added test for query feature.  (As well as reduplicated the round-trip test.)  Thanks for catching this, too.